### PR TITLE
Showing leaderboard precision value dynamically

### LINKED ISSUE #284: Add option to change leader-board decimal precision
Add option under settings tab in challenge page to change leader-board decimal precision by challenge host (**Note:** settings tab is only for the host user).

It is expected to add the front-end to be able to show the current `leaderbord decimal precision` value (refer field [here](https://github.com/Cloud-CV/EvalAI/blob/master/apps/challenges/models.py#L328)) and to add the option to change this field by the host user.

### PR DESCRIPTION
**Pull request by Abhi58**
Fix #284 :Add option to change leader-board decimal precision 

Fixes #284 
Changes proposed in this pull request:

-removed static leaderboard score value and added code to dynamically get the value from response obtained when a phase has been selected.

This is a work in progress PR

### CODE REVIEW

File: src/app/components/nav/header-static/header-static.component.html [hunk 1]
```diff
@@ -9,6 +9,7 @@
         <!-- show after authenticate user -->
         <li *ngIf="authService.isAuth && !isDash"><a class="waves-effect waves-dark main-header-link" (click) ="isDash = true" routerLink="/dashboard" routerLinkActive="active">Dashboard</a></li>
         <li *ngIf="authService.isAuth && isDash"><a class="waves-effect waves-dark main-header-link" routerLink="/challenges" routerLinkActive="active">All Challenges</a></li>
+        <li *ngIf="authService.isAuth"><a class="waves-effect waves-dark main-header-link" href="https://evalai.readthedocs.io/en/latest/" target="_blank">Docs</a></li>
```
Comment (Suryansh5545) [inline src/app/components/nav/header-static/header-static.component.html]: Why this change?

**Review by Suryansh5545 — CHANGES_REQUESTED** [review]

**Review by Suryansh5545 — CHANGES_REQUESTED** [review]
Can you Please add a field in settings so that the user or host can change the precision value

File: src/app/components/nav/header-static/header-static.component.html (hunk 1 above)
Comment (Abhi58) [reply src/app/components/nav/header-static/header-static.component.html]: I was comparing the original evalai with evalai-ngx and found that the there is a link to the documentation in the navbar of original evalai code. So, i added this line to for the same functionality in evalai-ngx too.

Comment (Suryansh5545) [reply src/app/components/nav/header-static/header-static.component.html]: Ok but this isn't related to this pr so undo it please

**Comment by Abhi58** [comment]
Sure! Thanks for the quick review.

File: src/app/components/nav/header-static/header-static.component.html (hunk 1 above)
Comment (Abhi58) [reply src/app/components/nav/header-static/header-static.component.html]: Okay. I will create a different pr for this!

**Comment by Sanji515** [comment]
Hey @Abhi58 On the [issue](https://github.com/Cloud-CV/EvalAI-ngx/issues/284#issue-561903296) you are expected to add option under `settings` tab of challenge page to change `leader-board decimal precision` by challenge host.

**Comment by Abhi58** [comment]
Sure @Sanji515 . I couldn't get the backend running from docker compose so, i compared with angularjs evali code for this commit. @Suryansh5545 has also mentioned the same. I will add the option under settings. Thanks for the quick review!

**Comment by Suryansh5545** [comment]
@Abhi58 Did you checked this issue #249 and tried the solution?

**Comment by Abhi58** [comment]
Hey @Suryansh5545. Yes, i checked the issue #249 and tried that. That way i was able to run the frontend but evalai-ngx only has code for front end. So, couldn't run the backend. I was trying to make it work by running original evalai and comparing the code on it with ngx.

**Comment by Abhi58** [comment]
Yesterday, i couldn't work as i had some assignments to submit. I will get back to working on this today. Sorry for the delay!

**Comment by Suryansh5545** [comment]
@Abhi58 Can you please post a log of the error you recieve when you try to run "docker-compose up" on evalai-ngx .  Also if you see the evalai-ngx repo pulls the evalai backend when you try to run it so you shouldn't require to run the original evalai repo

**Comment by Abhi58** [comment]
Hi @Suryansh5545 . I am attaching the screenshots of the logs. Please look into it. So, another thing is after i run the docker compose and then ng serve,the frontend is coming up. But, i am not able to access other functionalities like creating a new host which i feel is required to solve this issue. Could you tell me how  to run the backend without using docker compose? Thanks!

Screenshot from 2020-02-06 23-48-04
Screenshot from 2020-02-06 23-48-42

**Comment by Suryansh5545** [comment]
@Abhi58 This error is related to worker and angular so the django backend is not affected.  So just let the docker-compose up compile and then go to 127.0.0.1:8888 if you see a Hello world then the backend is active and running

**Comment by Shekharrajak** [comment]
It looks like @Abhi58 has not verified the code change because backend is not running. @Sanji515 can you please help @Abhi58  and share the link/step to run the backend server and login as dummy host user for dummy challenge?

**Comment by Shekharrajak** [comment]
@Abhi58 , are you following the steps given here: https://github.com/Cloud-CV/EvalAI-ngx#development-setup . You can also refer backend part here: https://github.com/Cloud-CV/EvalAI

**Comment by Suryansh5545** [comment]
@Shekharrajak Sir for some reason currently unknown the community is having problem running the EvalAI-ngx repository. So we are using a temporary solution which will run the front end and backend on different terminals . But for some reason @Abhi58 is unable to make it work.

**Comment by Shekharrajak** [comment]
> @Shekharrajak Sir for some reason currently unknown the community is having problem running the EvalAI-ngx repository. So we are using a temporary solution which will run the front end and backend on different terminals . But for some reason @Abhi58 is unable to make it work.

Oh I see! Let me know if there is open issue regarding this or discussion happened somewhere.

**Comment by Suryansh5545** [comment]
@Shekharrajak Sir this is the issue #249

**Comment by Shekharrajak** [comment]
> @Shekharrajak Sir this is the issue #249

Yeah! I was looking into that. Thanks!

**Comment by Abhi58** [comment]
Hi,

Mistakenly I had created this PR using master branch. I am closing this and created another PR with the fix: https://github.com/Cloud-CV/EvalAI-ngx/pull/287
